simulation/traffic_simulator/test/src/data_type/test_lanelet_pose.cpp

```diff
@@ -0,0 +1,390 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <scenario_simulator_exception/exception.hpp>
+#include <traffic_simulator/data_type/lanelet_pose.hpp>
+
+#include "../helper_functions.hpp"
+
+using CanonicalizedLaneletPose = traffic_simulator::lanelet_pose::CanonicalizedLaneletPose;
+
+int main(int argc, char ** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+
+/**
+ * @note Test constructor behavior with route_lanelets argument present when canonicalization function fails.
+ */
+TEST(CanonicalizedLaneletPose, CanonicalizedLaneletPose_withRoute_invalid)
+{
+  auto hdmap_utils = makeHdMapUtilsSharedPointer();
+  EXPECT_THROW(
+    CanonicalizedLaneletPose(
+      traffic_simulator::helper::constructLaneletPose(120576, 14.6356, 0.0), lanelet::Ids{},
+      hdmap_utils),
+    std::runtime_error);
```

Comment:
I think this function invokes `canonicalize` function, which throws a `common::SimulationError` or similar from `#include <scenario_simulator_exception/exception.hpp>`. If so, please change it

Reply:
changed to more obvious values as discussed